mm2src/coins/utxo/utxo_builder/utxo_coin_builder.rs

```diff
@@ -849,3 +849,13 @@ async fn wait_for_protocol_version_checked(client: &ElectrumClientImpl) -> Resul
     // Flatten `Result< Result<(), String>, String >`
     .flatten()
 }
+
+pub trait UtxoBlockSyncOps {
+    // Calculates the starting block height based on a given date and the current block height.
+    ///
+    /// # Arguments
+    /// * `date`: The date in seconds representing the desired starting date.
+    /// * `current_block_height`: The current block height at the time of calculation.
+    ///
+    fn calculate_starting_height_from_date(&self, date: u64, current_block_height: u64) -> Result<u64, String>;
```

Comment:
I think the right place for this is `UtxoCoinBuilderCommonOps` https://github.com/KomodoPlatform/komodo-defi-framework/blob/5e4ab780c776b2155f1bfd07de618a8f37a35d4b/mm2src/coins/utxo/utxo_builder/utxo_coin_builder.rs#L341 You can have a default implementation for it there too instead of implementing it for each coin as `avg_blocktime` is part of `UtxoCoinConf` https://github.com/KomodoPlatform/komodo-defi-framework/blob/3b9aa92e00c66ddddc859905cf92b4984d346f9b/mm2src/coins/utxo.rs#L567 and can be accessed inside `UtxoCoinBuilderCommonOps` using `self.conf()["avg_blocktime"]`

You should remove this too https://github.com/KomodoPlatform/komodo-defi-framework/blob/5e4ab780c776b2155f1bfd07de618a8f37a35d4b/mm2src/coins/z_coin.rs#L1067